File: test_archives/cpp_tests/batch5/test_function1.cpp
#include "../catch.hpp"
#include "../functions/function1.cpp"

TEST_CASE( "Tests for function1 class", "[function1]" ) {
    SECTION("Test for compute()") {
        function1 object("hello", 3, 4.5, true, {2, 4, 6});
        REQUIRE(object.compute() == "hello34.5True[2, 4, 6]");
    }
}

File: test_archives/py_tests/batch5/test_function1.py
from functions.function1 import function1
import pytest

@pytest.fixture
def setup_function1():
    return function1('hello', 3, 4.5, True, [2, 4, 6])

def test_compute(setup_function1):
    assert setup_function1.compute() == 'hello34.5True[2, 4, 6]'